### PR TITLE
Workflows

### PR DESCRIPTION
**Pull request by darrencauthon**
screenshot_122918_103215_am

### CODE REVIEW

File: app/views/events/index.html.erb [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
   function listenForPusherEvents() {
     var pusher = new Pusher('app_key', {
-      wsHost: '192.168.99.100',
+      wsHost: 'localhost',
```
Comment (calebcauthon) [inline app/views/events/index.html.erb]: if this worked for you, we prob need to make it configurable. docker doesn't run on "localhost" for me

File: app/models/workflow.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    steps.each_with_index { |x, i| x[:method] = (i == 0 ? :fire : :receive) }
+
+    steps.reduce(data) do |last_event, step_data|
+
+      step = step_data[:type].constantize.new
+
+      event = step.send(step_data[:method], last_event)
+
+      event.prior_event_id = last_event.id if last_event.is_a?(Event)
+      event.data = event.data || {}
```
Comment (calebcauthon) [inline app/models/workflow.rb]: makes sense. prior_event and data are part of the links between steps and generic to all steps. linking one event to another is applicable to all steps. 

`fire` and `receive` are the interfaces for event handlers, and we're only passing back the `last_event` to them. what happens inside of `fire` and `receive` are arbitrary.

File: app/models/event_handler.rb [hunk 3]
```diff
@@ -1,19 +1,11 @@
 class EventHandler
 
   def receive(event)
-    self.create_event message: "event ##{event.id} was fired", prior_event: event
+    Event.new message: "event ##{event.id} was fired"
```
Comment (calebcauthon) [inline app/models/event_handler.rb]: ha, I knew we wouldn't save here!

Comment (darrencauthon) [reply app/models/event_handler.rb]: But we weren't going to save in the controller, either.

Things may shift around till they find their proper home.

**Pull request merged**